Managers/SceneMan.cpp

```diff
@@ -357,6 +315,13 @@ int SceneMan::ReadProperty(const std::string_view &propName, Reader &reader)
 
                 // Assign the final ID to the material and register it in the palette
                 pNewMat->SetIndex(tryId);
+
+                // Ensure out-of-bounds material is unbreakable
+                // This is set as a very high value in data anyways, but not high enough ;)
+                if (tryId == g_MaterialOutOfBounds) {
```

Comment:
Why not just edit Base.rte/Materials.ini and up the strength of the `Default` Material?

Reply:
A comically large number isn’t the same here - the pathfinder treats a cost of float max as unpathable. Very high integrity isn’t strictly unpathable.

Reply:
Well yeah, fair enough, if they change they break everything anyway.

Anyway I still don't love this special behaviour. You say that float max is treated specially, but unless there's something I'm missing, any number could be treated specially so this argument doesn't hold. That is to say, we can define max possible material strength as whatever we wanted (though I agree float max is better than an arbitrary number).

Another problem here is that if someone wants to define another material that behaves like default material (i.e. in this case, is completely impassable to the pathfinder), they can't do that cause you're hardcoding it by index.

So yeah, if big number isn't viable, add support for the word "Max" or "Infinite" or something to the INI parsing so this behaviour is clear and properly accessible.

Reply:
The point is that it can't be any other number. If the value in the ini *isn't* float max, in this instance, everything breaks. We'd be exposing "max" to the ini, and then telling everyone, "yeah, this is in the ini, but you're not allowed to change this value otherwise the game will crash". What's the value in having that be exposed to data? It's literally just a crash-the-game switch.

Reply:
Either I'm doing a bad job of explaining what I mean, or you're doing a bad job of reading it all. Lemme try a bit more.
Though I do wanna say that
1. Adding the assert and requiring the value to be 0 is better than not having it, I'm happy with that change.
2. I was dumb earlier when I said to make it accept the string "Max" or "Infinite". There are a ton of places in the game where -1 means infinite, so you can simply treat -1 as float max (up to you if you wanna convert it to float max when you read it). The assert can stay even with this change I guess, though as you'll see at the bottom of this wall of text, it's a bit meaningless.
***
Moving onto my issues:

You say that only float max can tell the pathfinder we can't path, but that's because of this line of code at the bottom of pathfinder, right? `if (cost < std::numeric_limits<float>::max()) { ...`
Unless this isn't it, that means it is 100% arbitrary. I could change it to `if (static_cast<int>(cost) != 12345) { ...` and materials with SI 12345 would be unpathable. Obviously this is dumb, and if we're gonna have an arbitrary number, float max is the right choice, but I wanna correct what I see to be a false assertion on your part.
If I'm wrong about this, please let me know, I'd hate to be arguing based on a mistake haha.

More importantly than this nitpick, I think there's a core issue that you have to change your mindset on.
CC modding has always given excessive amounts of power to the modder, such that they can very easily brick their game if they please, or do things that don't really work. Maybe that's a bad idea in a big AAA game, but in a little indie game where half of modding is screwing around and seeing what happens, it can be fun and it's caused some interesting finds in the past (e.g. inverse glows, screen changes by palette swapping, etc.).
Here's a (rather tame) example - look how much fun Trez had with Y-wrapping a little while back. Y-wrapping is mostly broken, it barely works and is wonky as hell in real gameplay, so realistically it probably shouldn't be able to be turned on at the moment. All the same, Trez spent hours screwing around with it and had a ton of fun.
So yeah, If someone wants to make air have a non-zero SI and break their game, that's fine, it's a them problem. When they find weird shit happening, they can put it back how it should be (btw I actually just tried it out to see what happens, nothing significant changes cause we basically always ignore air).

The other part of this, and a point you completely ignored, is what if someone wants to make an unpathable material that actually looks like something not for testing? They can't, because there's no way for them to specify max SI.
E.g. somebody wants to make some maze activity that still allows digging. Or somebody wants to make a border around their map but not have it look like ass.

Below you can see the texture for the `Default` material, I think you'll agree that nobody's going to use this material in an actual map, because it's there solely for testing purposes. You're effectively adding functionality but making it never really usable, so the point you're fighting for kinda ends up being abstract nonsense anyway.
![Default](https://user-images.githubusercontent.com/7739094/204590822-659963bf-0cab-46ee-a1cb-8847f513a978.png)

Reply:
> You're effectively adding functionality but making it never really usable, so the point you're fighting for kinda ends up being abstract nonsense anyway.

This isn't a feature. This is not functionality I've added. The material is just the default material used for the pathing nodes so that their connections aren't traversable by default. This is the exact same as the old pathfinder behaviour, and is the reason I did things this way. I chose this material because it's the material with is already considered by the existing codebase as a special out-of-bounds index. It's not a material that's really meant to be used, and not a material that I added.

>  I could change it to if (static_cast<int>(cost) != 12345) { ... and materials with SI 12345 would be unpathable. Obviously this is dumb, and if we're gonna have an arbitrary number, float max is the right choice, but I wanna correct what I see to be a false assertion on your part.

The value itself is arbitrary, but the reason for choosing that value is not arbitrary.

Reply:
Just to note on github, this one has been figured out via pms in a way that we're both happy with :)

A few small changes need to be made to this stuff:
* -1 material integrity should be used to represent infinite in INI
* g_MaterialOutOfBoundsshould be renamed to g_MaterialTest or g_MaterialDefault.
* The cases where g_MaterialOutOfBounds are used should be replaced - the SLTerrain one should be use g_MaterialDefault or whatever for its default texture, the Atom one should probably just skip materials whose Integrity == float max
* A new out of bounds material (preferably with a flat black texture) should be added in INI and used for the job g_MaterialOutOfBounds is used for at the moment